### PR TITLE
feat: Token send heading component

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27562?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3219

## **Manual testing steps**

1. Go to this page...
2.
3.

#### Token with image
Screenshot 2024-10-01 at 14 48 07

#### Token without image but with symbol
Screenshot 2024-10-01 at 14 50 23

#### Token without image and symbol
Screenshot 2024-10-01 at 15 12 21

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/components/confirm/info/token-transfer/token-transfer.tsx [hunk 1]
```diff
@@ -1,5 +1,8 @@
+import React from 'react';
+import SendHeading from '../shared/send-heading/send-heading';
+
 const TokenTransferInfo = () => {
-  return null;
+  return <SendHeading />;
```
Comment (OGPoyraz) [inline ui/pages/confirmations/components/confirm/info/token-transfer/token-transfer.tsx:5]: Are we going to extend this component later or will it stay just importing this?

**Review by OGPoyraz — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/info/hooks/use-selected-token.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  getAllTokens,
+  getCurrentChainId,
+  getSelectedAccount,
+} from '../../../../../../selectors';
+
+export const useSelectedToken = (transactionMeta: TransactionMeta) => {
+  const selectedAccount = useSelector(getSelectedAccount);
+
+  const detectedTokens = useSelector(getAllTokens);
+  const chainId = useSelector(getCurrentChainId);
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/info/hooks/use-selected-token.ts:14]: We're actively trying to remove usages of the global network so can we instead extract this from the `transactionMeta`?

File: ui/pages/confirmations/components/confirm/info/hooks/use-selected-token.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  const selectedToken = detectedTokens?.[chainId]?.[
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/info/hooks/use-selected-token.ts:16]: Minor, but should be this be in a selector such as `selectWatchedToken` with arguments for chain and account?

File: ui/pages/confirmations/components/confirm/info/shared/send-heading/send-heading.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  return (
+    <Box
+      display={Display.Flex}
+      flexDirection={FlexDirection.Column}
+      justifyContent={JustifyContent.center}
+      alignItems={AlignItems.center}
+      paddingTop={4}
+    >
+      <AvatarToken
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/info/shared/send-heading/send-heading.tsx:43]: Minor, is this worth splitting into local `TokenImage` and `Value` components?

File: ui/pages/confirmations/components/confirm/info/hooks/use-token-values.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  const tokenBalance = useMemo(() => {
+    const tokenWithBalance = tokensWithBalances.find(
+      (token) =>
+        toChecksumHexAddress(token.address) ===
+        toChecksumHexAddress(transactionMeta?.txParams?.to as string),
+    );
+
+    if (!tokenWithBalance) {
+      return '';
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/info/hooks/use-token-values.ts:33]: Minor, is it cleaner and more explicit to use `undefined` here?

File: ui/pages/confirmations/components/confirm/info/hooks/use-token-values.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    return calcTokenAmount(tokenWithBalance.balance, tokenWithBalance.decimals);
+  }, [tokensWithBalances]);
+
+  const exchangeRate = useTokenExchangeRate(transactionMeta?.txParams?.to);
+
+  const fiatValue = useMemo(() => {
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/info/hooks/use-token-values.ts:41]: Minor, is a memo necessary for a single multiplication? Negligible performance cost?

File: ui/pages/confirmations/components/confirm/info/hooks/use-token-values.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useMemo } from 'react';
+import { calcTokenAmount } from '../../../../../../../shared/lib/transactions-controller-utils';
+import { toChecksumHexAddress } from '../../../../../../../shared/modules/hexstring-utils';
+import useTokenExchangeRate from '../../../../../../components/app/currency-input/hooks/useTokenExchangeRate';
+import { useFiatFormatter } from '../../../../../../hooks/useFiatFormatter';
+import { useTokenTracker } from '../../../../../../hooks/useTokenTracker';
+import { SelectedToken } from '../shared/selected-token';
+
+export const useTokenValues = (
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/info/hooks/use-token-values.ts:10]: Should we have unit tests for this hook also?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: ui/pages/confirmations/components/confirm/info/token-transfer/token-transfer.tsx (hunk 1 above)
Comment (pedronfigueiredo) [reply ui/pages/confirmations/components/confirm/info/token-transfer/token-transfer.tsx:5]: Yes, this work continues in the next PR https://github.com/MetaMask/metamask-extension/pull/27654

**Review by OGPoyraz — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**